controller/iteration_blackbox_test.go

```diff
@@ -593,6 +595,77 @@ func (rest *TestIterationREST) TestRootIterationCanNotStart() {
 	test.UpdateIterationBadRequest(rest.T(), svc.Context, svc, ctrl, ri.ID.String(), &payload)
 }
 
+func (rest *TestIterationREST) createIterations() (*app.IterationSingle, *account.Identity) {
+	sp, _, _, _, parent := createSpaceAndRootAreaAndIterations(rest.T(), rest.db)
+	_, err := rest.db.Iterations().Root(context.Background(), parent.SpaceID)
+	require.Nil(rest.T(), err)
+	parentID := parent.ID
+	name := uuid.NewV4().String()
+	ci := getChildIterationPayload(&name)
+	owner, err := rest.db.Identities().Load(context.Background(), sp.OwnerId)
+	require.Nil(rest.T(), err)
+	svc, ctrl := rest.SecuredControllerWithIdentity(owner)
+	// when
+	_, created := test.CreateChildIterationCreated(rest.T(), svc.Context, svc, ctrl, parentID.String(), ci)
+	// then
+	require.NotNil(rest.T(), created)
+	return created, owner
+}
+
+// TestIterationActivedByUser tests iteration should always be active when user sets it to active
+func (rest *TestIterationREST) TestIterationActivatedByUser() {
+	itr1, owner := rest.createIterations()
+	assert.Equal(rest.T(), false, *itr1.Data.Attributes.UserActive)
+	assert.Equal(rest.T(), true, *itr1.Data.Attributes.ActiveStatus) // iteration falls in timeframe, so iteration is active
+
+	startDate := time.Date(2017, 5, 17, 00, 00, 00, 00, time.UTC)
+	endDate := time.Date(2017, 6, 17, 00, 00, 00, 00, time.UTC)
+	userActive := true
+	payload := app.UpdateIterationPayload{
+		Data: &app.Iteration{
+			Attributes: &app.IterationAttributes{
+				StartAt:    &startDate,
+				EndAt:      &endDate,
+				UserActive: &userActive,
+			},
+			ID:   itr1.Data.ID,
+			Type: iteration.APIStringTypeIteration,
+		},
+	}
+	owner, errIdn := rest.db.Identities().Load(context.Background(), owner.ID)
+	require.Nil(rest.T(), errIdn)
+	svc, ctrl := rest.SecuredControllerWithIdentity(owner)
+	_, updated := test.UpdateIterationOK(rest.T(), svc.Context, svc, ctrl, itr1.Data.ID.String(), &payload)
+	assert.Equal(rest.T(), iteration.IterationActive, *updated.Data.Attributes.ActiveStatus) // iteration doesnot fall in timeframe yet userActive is true so iteration is active
+}
+
+// TestIterationActivatedByTimeframe tests
+// 1. Iteration should be active when it is in timeframe
+// 2. Iteration should not be active when it is outside the timeframe
+func (rest *TestIterationREST) TestIterationActivatedByTimeframe() {
```

Comment:
You can avoid the repetitive test setup code in `TestIterationActivatedByUser` and `TestIterationActivatedByTimeframe` if you'd use a subtest.

Reply:
changed it here d99ce556bf9ea4b209176790e2a0af9381a88669